.github/workflows/_push_image.yaml

```diff
@@ -0,0 +1,78 @@
+name: Push images
+
+# This is a reusable workflow that is called from any other workflow that needs 
+# to build and push the configurator image to DockerHub
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Version as is going to use for the binary'
+        type: string
+        required: true
+      image-tag:
+        description: 'Tag as is going to use for the image'
+        type: string
+        required: true
+      image-overwrite-latest:
+        description: 'If enabled, an extra tag `latest` will be pushed apart from the defined tag'
+        type: boolean
+        required: false
+        default: false
+
+jobs:
+  push-image:
+    name: Build and Push images to Docker Hub
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_IMAGE_NAME: newrelic/newrelic-prometheus-configurator
+      DOCKER_PLATFORMS: "linux/amd64,linux/arm64,linux/arm" # Must be consistent with make build-multiarch
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build integration
+        env:
+          TAG: ${{ inputs.version }}
+        run: make build-multiarch
+      
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.FSI_DOCKERHUB_USERNAME }}
+          password: ${{ secrets.FSI_DOCKERHUB_TOKEN }}
+
+      - name: Build and load x64 image for security scanning
+        # We need to build a single-arch image again to be able to --load it into the host
+        run: |
+          docker buildx build --load --platform=linux/amd64 \
+            -t $DOCKER_IMAGE_NAME:ci-scan \
+            .
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@0.7.1
+        with:
+          image-ref: '${{ env.DOCKER_IMAGE_NAME }}:ci-scan'
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          severity: 'HIGH,CRITICAL'
+
+      - name: Push image
+        if: ${{ ! inputs.image-overwrite-latest }}
+        run: |
+          docker buildx build --push --platform=$DOCKER_PLATFORMS \
+            -t $DOCKER_IMAGE_NAME:${{ inputs.image-tag }} \
+            .
+
+      - name: Push image and overwrite 'latest'
+        if: ${{ inputs.image-overwrite-latest }}
+        run: |
+          docker buildx build --push --platform=$DOCKER_PLATFORMS \
+            -t $DOCKER_IMAGE_NAME:${{ inputs.image-tag }} \
+            -t $DOCKER_IMAGE_NAME:latest \
+            .
```

Comment:
Could we have a different workflow for the pre-release and the release?

I mean, while with the prerelease we upload a version like `v1-pre` in the release, instead of retesting and compiling everything again we could simply:

```
docker pull ${version}-pre
docker tag ${version}-pre $version
docker push $version
```

That would make the release way faster. WDYT?

Reply:
I though about this and tried to implemented but this doesn't  work 
```
docker pull ${version}-pre
docker tag ${version}-pre $version
docker push $version
```
since these are multi-arch images. 

On the other hand , pre-release has been partially removed for now, meaning that on the semi-automated release workflow we will release directly.

Reply:
Makes sense. For the next iteration... Thanks!